subsys/mpsl/cx/bluetooth/mpsl_cx_bluetooth.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ *   This file implements a generic bluetooth external radio coexistence
+ *   interface.
+ */
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <devicetree.h>
+#include <nrfx.h>
+#include <nrfx_ppi.h>
+#include <nrfx_gpiote.h>
+#include <mpsl_coex.h>
+
+#define COEX_TIMER NRF_TIMER1
+
+#if DT_HAS_COMPAT_STATUS_OKAY(sdc_radio_coex_three_wire)
+#define COEX_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(sdc_radio_coex_three_wire)
+#else
+#define COEX_NODE DT_INVALID_NODE
+#error No "sdc-radio-coex-three-wire"-compatible (and enabled) nodes found
+#endif
+
+#if !IS_ENABLED(CONFIG_SOC_SERIES_NRF52X)
+#error Bluetooth coex is only supported on the nRF52 series
+#endif
+
+#define MPSL_COEX_BT_GPIO_POLARITY_GET(dt_property)                                                \
+	((GPIO_ACTIVE_LOW & DT_GPIO_FLAGS(COEX_NODE, dt_property)) ? false : true)
+
+static volatile bool coex_enabled;
+
+static void coex_enable_callback(void)
+{
+	coex_enabled = true;
+}
+
+int mpsl_cx_bt_interface_3wire_config_set(void)
+{
+	nrfx_err_t err = NRFX_SUCCESS;
+	mpsl_coex_gpiote_cfg_t *gpiote_cfg;
+	nrf_ppi_channel_t ppi_channel;
+	mpsl_coex_if_t coex_if_bt;
+	mpsl_coex_802152_3wire_gpiote_if_t *coex_if = &coex_if_bt.interfaces.coex_3wire_gpiote;
+
+	/* Allocate GPIOTE and PPI channels for the REQUEST line */
+	gpiote_cfg = &coex_if->request_cfg;
+	if (nrfx_gpiote_channel_alloc(&gpiote_cfg->gpiote_ch_id) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+	if (nrfx_ppi_channel_alloc(&ppi_channel) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+	gpiote_cfg->gpio_pin = DT_GPIO_PIN(COEX_NODE, req_gpios);
```

Comment:
Please double check this file for errors where a pin and PSEL value are conflated. I'm not saying there are some, but this type of code is generally suspicious since I don't see the GPIO controller in `req-gpios` mentioned anywhere. (I left a comment with more details on PSEL vs. pin elsewhere in this PR.)